diff --git a/contrib/llvm/tools/lld/ELF/Arch/X86_64.cpp b/contrib/llvm/tools/lld/ELF/Arch/X86_64.cpp
index 3db391e9f..802088c08 100644
--- a/contrib/llvm/tools/lld/ELF/Arch/X86_64.cpp
+++ b/contrib/llvm/tools/lld/ELF/Arch/X86_64.cpp
@@ -108,6 +108,8 @@ RelExpr X86_64<ELFT>::getRelExpr(RelType Type, const Symbol &S,
     return R_GOT_PC;
   case R_X86_64_NONE:
     return R_NONE;
+  case R_X86_64_RELATIVE:
+    return R_HINT;
   default:
     return R_INVALID;
   }
