--- sys/vm/vm_map.h.orig	2020-08-14 12:10:43.380509000 +0200
+++ sys/vm/vm_map.h	2020-08-14 11:44:16.398386000 +0200
@@ -206,6 +206,7 @@
 	vm_flags_t flags;		/* flags for this vm_map */
 	vm_map_entry_t root;		/* Root of a binary search tree */
 	pmap_t pmap;			/* (c) Physical map */
+	vm_offset_t anon_loc;
 	int busy;
 };

@@ -214,6 +215,9 @@
  */
 #define MAP_WIREFUTURE		0x01	/* wire all future pages */
 #define	MAP_BUSY_WAKEUP		0x02
+#define	MAP_IS_SUB_MAP		0x04	/* has parent */
+#define	MAP_ASLR		0x08	/* enabled ASLR */
+#define	MAP_ASLR_IGNSTART	0x10

 #ifdef	_KERNEL
 #if defined(KLD_MODULE) && !defined(KLD_TIED)
